File: spotify-backend/wishlistservice/src/main/java/com/spotify/wishlistservice/model/Album.java
package com.spotify.wishlistservice.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Album {
    private String albumType;
    private int totalTracks;
    private List<String> availableMarkets;
    private ExternalUrls externalUrls;
    private String href;
    private String id;
    private List<Image> images;
    private String name;
    private String releaseDate;
    private String releaseDatePrecision;
    private Restrictions restrictions;
    private String type;
    private String uri;
    private List<Artist> artists;
    // Other fields and methods

    // Getters and setters
}

File: spotifyApp/src/app/model/Album.ts
import { ExternalUrls } from "./ExternalUrls";
import { Artist } from "./Artist";
import { Image } from "./Image";

export class Album{
    albumType!: string;
    artists!: Artist[];
    externalUrls!: ExternalUrls;
    images!: Image[];
    id!: string;
    name!: string;
    releaseDate!: string;
    type!: string;
    /*
    constructor(
        public albumType: string,
        public artists: Artist[],
        public externalUrls: ExternalUrls,
        public images: Image[],
        public id: string,
        public name: string,
        public releaseDate: string,
        public type: string
    ) {}
    */
}